app/javascript/packages/document-capture/components/in-person-location-step.tsx

```diff
@@ -30,9 +30,62 @@ interface FormattedLocation {
   weekdayHours: string;
 }
 
+interface PostOptions {
+  /**
+   * Whether to send the request as a JSON request. Defaults to inclusion.
+   */
+  json: boolean;
+
+  /**
+   * Whether to include CSRF token in the request. Defaults to inclusion.
+   */
+  csrf: boolean;
+
+  /**
+   * Optional. HTTP verb used. If excluded, will fetch as a GET.
+   */
+  method: string;
+}
+
+const getCsrfToken = () =>
+  document.querySelector<HTMLMetaElement>('meta[name="csrf-token"]')?.content;
+
+const request = async (
+  url: string,
+  body: BodyInit | object,
+  options: Partial<PostOptions> = {},
+) => {
+  const headers: HeadersInit = {};
+  const defaultOptions = {
+    csrf: true,
+    json: true,
+    ...options,
+  };
+
+  if (defaultOptions.csrf) {
+    const csrf = getCsrfToken();
+    if (csrf) {
+      headers['X-CSRF-Token'] = csrf;
+    }
+  }
+
+  if (defaultOptions.json) {
+    headers['Content-Type'] = 'application/json';
+    body = JSON.stringify(body);
+  }
+
+  const response = await window.fetch(url, {
+    ...(defaultOptions.method ? { method: defaultOptions.method } : {}),
```

Comment:
Wonder if it would work just as well to assign the method value directly, and trust that `fetch` would ignore the `undefined` value if it's not assigned.

```suggestion
    method: mergedOptions.method,
```